uint8 agentIdx
int8 TaskType
bool newReward
